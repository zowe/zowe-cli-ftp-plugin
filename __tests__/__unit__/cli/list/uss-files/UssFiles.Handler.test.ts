/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import listDatasetHandler from "../../../../../src/cli/list/uss-files/UssFiles.Handler";
import TestUtils from "../../TestUtils";

describe("List uss files handler", () => {

    it("should return no uss file if the uss file is not found.", async () => {
        const handler = new listDatasetHandler();
        const files: any[] = [];

        const mockResponse = TestUtils.getMockResponse();
        const mockParams: any = {
            arguments: {
                directory: "/u/user"
            },
            connection: {
                listDataset: jest.fn().mockReturnValue(Promise.resolve(files))
            },
            response: mockResponse
        };
        await handler.processFTP(mockParams);
        expect(mockResponse.data.setMessage.mock.calls[0][0]).toBe("Listed files in uss directory %s");
        expect(mockResponse.data.setMessage.mock.calls[0][1]).toBe("/u/user");
        expect(mockResponse.data.setObj.mock.calls[0][0]).toMatchSnapshot();
        expect(mockResponse.format.output.mock.calls[0][0]).toMatchSnapshot();
    });

    it("should return correct message if the uss file is found.", async () => {
        const handler = new listDatasetHandler();
        const files: any[] = [
            {
                name: "ds1"
            },
            {
                name: "ds2"
            }
        ];

        const mockResponse = TestUtils.getMockResponse();
        const mockParams: any = {
            arguments: {
                directory: "/u/user"
            },
            connection: {
                listDataset: jest.fn().mockReturnValue(Promise.resolve(files))
            },
            response: mockResponse
        };
        await handler.processFTP(mockParams);
        expect(mockResponse.data.setMessage.mock.calls[0][0]).toBe("Listed files in uss directory %s");
        expect(mockResponse.data.setMessage.mock.calls[0][1]).toBe("/u/user");
        expect(mockResponse.data.setObj.mock.calls[0][0]).toMatchSnapshot();
        expect(mockResponse.format.output.mock.calls[0][0]).toMatchSnapshot();
    });

    it("should return correct files if the file name pattern is specified.", async () => {
        const handler = new listDatasetHandler();
        const files: any[] = [
            { name: "ds1" },
            { name: "ds2" },
            { name: "dt1" },
            { name: "dt2" }
        ];

        const mockResponse = TestUtils.getMockResponse();
        const mockParams: any = {
            arguments: {
                directory: "/u/user/ds*"
            },
            connection: {
                listDataset: jest.fn().mockReturnValue(Promise.resolve(files))
            },
            response: mockResponse
        };
        await handler.processFTP(mockParams);
        expect(mockResponse.data.setMessage.mock.calls[0][0]).toBe("Listed files in uss directory %s");
        expect(mockResponse.data.setMessage.mock.calls[0][1]).toBe("/u/user/ds*");
        expect(mockResponse.data.setObj.mock.calls[0][0]).toMatchSnapshot();
        expect(mockResponse.format.output.mock.calls[0][0]).toMatchSnapshot();

        mockParams.arguments.directory = "/u/user/*1";
        await handler.processFTP(mockParams);
        expect(mockResponse.data.setMessage.mock.calls[1][0]).toBe("Listed files in uss directory %s");
        expect(mockResponse.data.setMessage.mock.calls[1][1]).toBe("/u/user/*1");
        expect(mockResponse.data.setObj.mock.calls[1][0]).toMatchSnapshot();
        expect(mockResponse.format.output.mock.calls[1][0]).toMatchSnapshot();

        mockParams.arguments.directory = "/u/user/d*2";
        await handler.processFTP(mockParams);
        expect(mockResponse.data.setMessage.mock.calls[2][0]).toBe("Listed files in uss directory %s");
        expect(mockResponse.data.setMessage.mock.calls[2][1]).toBe("/u/user/d*2");
        expect(mockResponse.data.setObj.mock.calls[2][0]).toMatchSnapshot();
        expect(mockResponse.format.output.mock.calls[2][0]).toMatchSnapshot();

        mockParams.arguments.directory = "/u/user/*";
        await handler.processFTP(mockParams);
        expect(mockResponse.data.setMessage.mock.calls[3][0]).toBe("Listed files in uss directory %s");
        expect(mockResponse.data.setMessage.mock.calls[3][1]).toBe("/u/user/*");
        expect(mockResponse.data.setObj.mock.calls[3][0]).toMatchSnapshot();
        expect(mockResponse.format.output.mock.calls[3][0]).toMatchSnapshot();
    });
});
