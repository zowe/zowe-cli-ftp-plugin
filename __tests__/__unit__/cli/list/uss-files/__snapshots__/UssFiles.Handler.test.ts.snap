// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List uss files handler should return correct files if the file name pattern is specified. 1`] = `
Array [
  Object {
    "name": "ds1",
  },
  Object {
    "name": "ds2",
  },
]
`;

exports[`List uss files handler should return correct files if the file name pattern is specified. 2`] = `
Object {
  "fields": Array [
    "name",
    "size",
    "owner",
    "group",
    "permissions",
  ],
  "format": "table",
  "output": Array [
    Object {
      "name": "ds1",
    },
    Object {
      "name": "ds2",
    },
  ],
}
`;

exports[`List uss files handler should return correct files if the file name pattern is specified. 3`] = `
Array [
  Object {
    "name": "ds1",
  },
  Object {
    "name": "dt1",
  },
]
`;

exports[`List uss files handler should return correct files if the file name pattern is specified. 4`] = `
Object {
  "fields": Array [
    "name",
    "size",
    "owner",
    "group",
    "permissions",
  ],
  "format": "table",
  "output": Array [
    Object {
      "name": "ds1",
    },
    Object {
      "name": "dt1",
    },
  ],
}
`;

exports[`List uss files handler should return correct files if the file name pattern is specified. 5`] = `
Array [
  Object {
    "name": "ds2",
  },
  Object {
    "name": "dt2",
  },
]
`;

exports[`List uss files handler should return correct files if the file name pattern is specified. 6`] = `
Object {
  "fields": Array [
    "name",
    "size",
    "owner",
    "group",
    "permissions",
  ],
  "format": "table",
  "output": Array [
    Object {
      "name": "ds2",
    },
    Object {
      "name": "dt2",
    },
  ],
}
`;

exports[`List uss files handler should return correct files if the file name pattern is specified. 7`] = `
Array [
  Object {
    "name": "ds1",
  },
  Object {
    "name": "ds2",
  },
  Object {
    "name": "dt1",
  },
  Object {
    "name": "dt2",
  },
]
`;

exports[`List uss files handler should return correct files if the file name pattern is specified. 8`] = `
Object {
  "fields": Array [
    "name",
    "size",
    "owner",
    "group",
    "permissions",
  ],
  "format": "table",
  "output": Array [
    Object {
      "name": "ds1",
    },
    Object {
      "name": "ds2",
    },
    Object {
      "name": "dt1",
    },
    Object {
      "name": "dt2",
    },
  ],
}
`;

exports[`List uss files handler should return correct message if the uss file is found. 1`] = `
Array [
  Object {
    "name": "ds1",
  },
  Object {
    "name": "ds2",
  },
]
`;

exports[`List uss files handler should return correct message if the uss file is found. 2`] = `
Object {
  "fields": Array [
    "name",
    "size",
    "owner",
    "group",
    "permissions",
  ],
  "format": "table",
  "output": Array [
    Object {
      "name": "ds1",
    },
    Object {
      "name": "ds2",
    },
  ],
}
`;

exports[`List uss files handler should return no uss file if the uss file is not found. 1`] = `Array []`;

exports[`List uss files handler should return no uss file if the uss file is not found. 2`] = `
Object {
  "fields": Array [
    "name",
    "size",
    "owner",
    "group",
    "permissions",
  ],
  "format": "table",
  "output": Array [],
}
`;
